docs/tutorials/basics/concepts.md

```diff
@@ -0,0 +1,128 @@
+synonymous with “HPO algorithm” in Syne Tune. 
+
+### Tuner
+Tuner moderates the HPO process by interacting with _scheduler_ and _backend_. 
+It queries the _scheduler_ for a trial when a worker is free, and pass the trial to the backend
+for execution.  
+
+### Scheduler
+In Syne Tune, HPO algorithms are called _schedulers_. 
+Schedulers search for a new, most promising configuration and suggest it as a new trial to _tuner_. 
+Some schedulers may decide to resume a paused trial instead of suggesting a new one.
+Schedulers are also in charge of stopping running trials. Syne Tune supports many schedulers including multi-fidelity ones and asynchronous ones.
+
+### Backend
+The backend module is responsible for starting, stopping, pausing and resuming trials and accessing
+results and trial statuses. Syne Tune supports three execution backends to facilitate experimentations: local backend, SageMaker backend, and simulated backend. 
+To explain these backends, recall that any HPO job consists of two main scripts: a tuner script where HPO loop runs and next configurations from search space is selected, and a training script where the machine learning model (whether it is a neural network or a tree-based model) is trained based on the selected hyperparameter configuration from tuner.
+The execution backend runs these scripts; and depending on whether they are executed on one or multiple machines, the execution backend changes.  
+
+Local backend runs the training job locally with respect to where tuner job is running. That is if both tuner script and training script run on the same machine (whether it is your local machine or in cloud) it is local backend. 
+Figures \ref and \ref demonstrates local backend, where in Figure \ref{} both scripts are executed on local machine, and in Figure \ref{} they are executed on a SageMaker instance in cloud. 
+
+
+<table>
+  <tr>
+    <td style="border-color: white; width: 600px"><img src="./img/local1.png" width="200" />
+    <figcaption>Local backend on a local machine</figcaption>
+    </td>
+    <td style="border-color: white"><img src="./img/local2.png" width="530" />
+    <figcaption> Local backend when running on a SageMaker instance on cloud</figcaption>
+    </td>
+  </tr>
+</table>
+
+
+
+Local backend evaluates trials concurrently on a single machine by using subprocesses.
+SyneTune support rotating multiple GPUs on the machine, assigning the next trial to the least
+busy GPU, e.g. the GPU with the smallest amount of trials currently running. 
+
+Local backend suffers from two shortcomings; first, it limits the number of trials that can run concurrently.
+Second, it falls short in training neural network which require many GPUs, or possibly distributed across several nodes
+
+
+SageMaker backend covers this gap by executing training script in cloud and specifically in an Amazon SageMaker instance while the tuner script runs either from your local machine or from another SageMaker instance in cloud. 
```

Comment:
In general, I think this first section is getting too long. It is meant just to introduce the basic concepts and as a glossary.

It should not present any code or launcher scripts. For that, please just refer to the scripts used further down.

Note: I am refactoring this material for the sphinx docs, and this will have scripts being inline (instead of files), so more or less what you are doing here, but at the right place (step by step).

Reply:
done. removed the code snippets.